### PR TITLE
Docs: fix saml role addition partial.

### PR DESCRIPTION
**Pull request by alexfornuto**
In our partial file for adding roles to users, the SAML instructions have us create a local file of the saml connector resource. Without the `--with-secrets` flag an empty `private_key` value is set in the file. Once the connector is recreated authentication fails.

This PR add the `--with-secrets` flag, and warns the user that this copies sensitive data.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: docs/pages/includes/add-role-to-user.mdx [hunk 1]
```diff
@@ -91,6 +91,12 @@ Apply your changes:
 $ tctl create -f saml.yaml
 ```
 
+<Admonition type="warning">
+Note the `--with-secrets` flag in the `tctl get` command. This adds the value of
+`spec.signing_key_pair.private_key`. to `saml.yaml`. This is a sensitive value,
```
Comment (ptgott) [inline docs/pages/includes/add-role-to-user.mdx:96]:
```suggestion
`spec.signing_key_pair.private_key` to `saml.yaml`. This is a sensitive value,
```

File: docs/pages/includes/add-role-to-user.mdx [hunk 2]
```diff
@@ -91,6 +91,12 @@ Apply your changes:
...
+so take precautions when creating this file and remove it after updating the resource.
```
Comment (ptgott) [inline docs/pages/includes/add-role-to-user.mdx:97]: This is possibly implied, but just to triple-check: does "it" refer to the file or the value?

Comment (alexfornuto) [reply docs/pages/includes/add-role-to-user.mdx:97]: The file.

Comment (alexfornuto) [reply docs/pages/includes/add-role-to-user.mdx:97]: Setting script file permissions makes sense in a trusted environment. Removing the file makes sense in others, but doesn't work in a compromised environment in the first place. By being less specific we're leaving it to the reader to determine what security measures are appropriate for their environment. By saying what precautions are acceptable we open ourselves up to liability.

**Review by ptgott — APPROVED** [review]

**Comment by tigrato** [comment]
@alexfornuto 

I added it because our docs were wrong and it resulted in a S1 incident (sorry for jumping your PR but wasn't aware of it).

I think it makes sense to add the warning everywhere but I feel we need to improve the UX to avoid it happening again. i.e. allowing patches without exporting the private_key.

Now that `tctl edit ` is a thing, we could start advocating it to avoid having users exporting and sharing files with sensitive data.

**Comment by alexfornuto** [comment]
@ptgott please take a look at the last two commits.

**Review by ptgott — APPROVED** [review]

**Pull request merged**